eda/geo_feature_engineering.py

```
@@ -0,0 +1,244 @@
+        # 생성된 피처를 DF에 추가
+        feature_df = pd.DataFrame(features, columns=['academy_500m', 'academy_1km', 'academy_access'], index=df.index).fillna(0)
+        df = pd.concat([df, feature_df], axis=1)
+    except Exception as e: print(f"학원 피처 생성 중 오류 발생: {e}"); pass
+    return df
+
+def add_traffic_features(df, traffic_df, x_col, y_col):
+    """통행불편지역 관련 피처를 추가."""
+    try:
+        traf_x, traf_y = find_coord_columns(traffic_df)
+        if not traf_x or not traf_y or len(traffic_df.dropna(subset=[traf_x, traf_y])) == 0: 
+            print("경고: 유효한 통행불편지역 데이터 또는 컬럼 없어. 통행 피처 생성 건너뜀."); return df
```

Ruff (0.12.2)

Multiple statements on one line (semicolon)

(E702)